src/operations/execute_operation.ts

```diff
@@ -118,6 +118,14 @@ export async function executeOperation<
     );
   }
 
+  const timeoutContext = TimeoutContext.create({
```

Comment:
My intention was that we construct exactly one `TimeoutContext` per operation with a lifetime equal to that of the operation. The first convergent place we have access to both the client and operation options would be in `executeOperation`, but the absolute first place is in each of our operation helper methods (insertOne, updateOne, etc).

I am open to updating our `AbstractOperation` constructor to accept the client options we'd need to construct the `TimeoutContext` and updating these methods since that would be cleaner and make clear that each operation "owns" its `TimeoutContext`.

Wouldn't constructing the `TimeoutContext` in `resolveOptions` lead to multiple `TimeoutContext` objects being constructed that likely wouldn't be used.

Reply:
> I am open to updating our AbstractOperation constructor to accept the client options we'd need to construct the TimeoutContext and updating these methods since that would be cleaner and make clear that each operation "owns" its TimeoutContext.

Yeah, you're right that resolveOptions would lead to instantiation of timeouts we don't need, specifically when we construct dbs and collections.  I'm not sure an operation owning a timeout context is the right choice either - I think that cursors / change streams will need to "own" their timeout context and they'll perform multiple operations inside the same context.  Maybe we can leave it as-is for now and reconsider once we've implemented CSOT for cursors.